beacon-chain/sync/fork_watcher.go

```diff
@@ -95,7 +105,15 @@ func (s *Service) deregisterFromPastFork(currEpoch types.Epoch) error {
 		if !s.subHandler.digestExists(prevDigest) {
 			return nil
 		}
-		s.unregisterPhase0Handlers()
+		prevFork, err := forks.Fork(epochBeforeFork)
+		if err != nil {
+			return errors.Wrap(err, "Failed to determine previous epoch fork data")
+		}
+
+		switch prevFork.Epoch {
+		case params.BeaconConfig().GenesisEpoch:
+			s.unregisterPhase0Handlers()
+		}
```

Comment:
Switch with a single case? I presume this is a placeholder for Altair unregistrations?

Reply:
There are no altair unregistrations, rpc methods stay the same in bellatrix

Reply:
Isn't it more natural to write
```
if prevFork.Epoch == params.BeaconConfig().GenesisEpoch {
    s.unregisterPhase0Handlers()
}
```

Reply:
We will have new methods most likely added/removed in newer forks. The switch brings it inline with fork handling code across prysm.